shop/src/app/app-routing.module.ts

```diff
@@ -4,12 +4,19 @@ import { ProductsListComponent } from './components/containers/products-list/pro
 import { ProductsDetailsComponent } from './components/containers/products-details/products-details.component';
 import { ShoppingCartDetailsComponent } from './modules/shopping-cart/components/containers/shopping-cart-details/shopping-cart-details.component';
 import { Title } from '@angular/platform-browser';
+import { BadRouteComponent } from './components/bad-route/bad-route.component';
+import { AddProductComponent } from './components/containers/add-product/add-product.component';
+import { EditProductComponent } from './components/containers/edit-product/edit-product.component';
 
 const routes: Routes = [
   { path: '', component: ProductsListComponent },
   { path: 'product-list', component: ProductsListComponent, title: 'Products' },
-  { path: 'product-details/product0', component: ProductsDetailsComponent, title: 'Product Details' },
-  { path: 'shopping-cart', component: ShoppingCartDetailsComponent, title: 'Cart' }
+  { path: 'product-details/:id', component: ProductsDetailsComponent, title: 'Product Details' },
+  { path: 'shopping-cart', component: ShoppingCartDetailsComponent, title: 'Cart' },
+  { path: 'add-product', component: AddProductComponent, title: 'Add Product' },
+  { path: 'edit-product/:id', component: EditProductComponent, title: 'Edit Product' },
+  { path: 'product-details/:id/edit-product', component: EditProductComponent, title: 'Edit Product' },
```

Comment:
2 different paths pointing to the same compoment. Are both used?

Reply:
Deleted the first one